in_toto/runlib.py

```diff
@@ -374,8 +374,11 @@ def in_toto_run(name, material_list, product_list,
     log.info("Recording materials '{}'...".format(", ".join(material_list)))
   materials_dict = record_artifacts_as_dict(material_list)
 
-  log.info("Running command '{}'...".format(" ".join(link_cmd_args)))
-  byproducts, return_value = execute_link(link_cmd_args, record_byproducts)
+  if link_cmd_args:
+    log.info("Running command '{}'...".format(" ".join(link_cmd_args)))
+    byproducts, return_value = execute_link(link_cmd_args, record_byproducts)
+  else:
+    byproducts, return_value = 0, 0
```

Comment:
Not sure what `byproducts` and `return_value` should be if no command was executed. I wouldn't use `0` though, because that's a actually common return value.

In `in_toto_record`, where we also generate a Link without having executed a command, we do `byproducts={}` and `return_value=None`. What do you think?

Reply:
I think `byproduct={}` and `return_value=None` is better. I'll switch to that.